Documentation/ApiOverview/ContentSecurityPolicy/Index.rst

```diff
@@ -46,7 +59,184 @@ need to be enabled.
+
+Nonces
+------
+
+*   There are possible exemptions to directives for specific content created on specific
+    pages created by TYPO3 in your frontend (or backend modules). To verify, that these
+    exemptions are valid in a policy, a so-called "Nonce" (unique hash, a "**n**umber used **once**")
+    is created (details on `https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/nonce`__).
+
+    *   TYPO3 can manage these Nonces and apply them were configured.
+    *   Nonces are retrieved from :php:`\TYPO3\CMS\Core\Security\ContentSecurityPolicy\ConsumableNonce`
+        and will be used for any directive within the scope of a single HTTP request.
+    *   More details are covered in :ref:`content-security-policy-nonce`.
+
+..  _content-security-policy-terminology-violations:
+
+Policy violations and reporting
+-------------------------------
+
+*   When a webpage with activated policies is shown in a client's browser, each HTML tag
+    violating the policy will not be interpreted by the browser.
+
+*   Depending on a configuration of a possible "Report", such violations can be submitted
+    back to the server and be evaluated there. TYPO3 provides such an interface to receive
+    and display reports in a backend module.
+
+*   Policies can be declared with "dispositions", to indicate how they are handled.
+    "Enforce" means that a policy is in effect, and "Report-Only" allows to only pretend
+    a policy is in effect, to gather knowledge about possible improvements of a
+    webpage's output. Both dispositions can be set independently in TYPO3.
+
+*   All active rules can be seen in the backend configuration section, see
+    :ref:`content-security-policy-backend-rules`.
+
+..  _content-security-policy-example:
+
+Example scenario
+================
+
+Let's define a small real-world scenario:
+
+*   You have one TYPO3 installation with two sites (frontend), `example.com` and `example.org`.
+*   You have created custom backend modules for some distinct functionality.
+*   `example.com` is a site where your editors fully control all frontend content, and they
+    want to have full flexibility of what and how to embed. You use a CDN network to deliver
+    your own large assets, like videos.
+*   `example.org` is a community-driven site, where users can manage profiles and post chats,
+    and where you want to prevent exploits on your site. Some embedding to a set of allowed
+    web services (YouTube, Google Analytics) must be possible.
+
+So you need to take care of security measures, and find a pragmatic way how to allow foreign
+content (like YouTube, widgets, tracking codes, assets)
+
+Specifically you want to to set these rules, as an example:
+
+Rules for example.com (editorial)
+---------------------------------
+
+*    :html:`<iframe>` to anyhwere should be allowed
+*    :html:`<img>` sources to anywhere should be allowed
+*    :html:`<script>` sources to `cdn.example.com` and `*.youtube.com` and `*.google.com`
```

Comment:
GA needs other sources than "*.google.com":
https://content-security-policy.com/examples/google-analytics/

YouTube is allowed by default in TYPO3 frontend.

Reply:
This puts me in a bit of a pickle. I don't really have the time to dig into all the variations for Analytics/Youtube and really only wanted to point out as an example. I've now added an disclaimer that the example is just for demo purposes and not actual URIs... does this work for you!?

Reply:
It's perfect for showing the idea and example 👌

Reply:
Fine for me.